ls1intum/Artemis
# Fix Automatic Saves for "Edit in editor"

### Checklist

### Motivation and Context
The code editor for the instructor's "Edit in editor" mode is unusable in version `5.0.0`.

### Description
- Removed the `PendingChangesGuard` which caused type errors because `code-editor-instructor-base-container` component is no longer implements `ComponentCanDeactivate` (#3617).
- Added missing auto save when the repository of the code editor is changed.

### Steps for Testing
1. Open the Code Editor of "Edit in editor" (as instructor for a programming exercise in course management)
2. Play around with changing files, switch the repository, close the tab, navigate somewhere else, ... and make sure that the changes in the files are not lost (for that you need to remember the changes). You can do this pretty quickly as well to avoid the 30 s interval autosave.
